### ISSUE TITLE
auto archive or archive every X hours?

### ISSUE THREAD

**Issue by zeltak** [ivan-lednev/obsidian-task-archiver#82]
Hi and thank you for all your amazing work!

I may have missed it but can one define to auto archive once the task is marked is done or maybe auto archive every X hours?

thx so much!!

Z

**Comment by ChayseSmith** [comment]
I second this request, I was expecting the archive function to run the instant I complete a task. That would be ideal in my situation. The plugin you've created is fantastic however and still saves me a lot of time.

**Comment by terafin** [comment]
Yes please :)

**Comment by ivan-lednev** [comment]
Hello,

There is already a command called `Toggle task done and archive it`. Once you assign it to a shortcut (like `ctrl + Enter`) this gets you 80% there.

**Comment by ChayseSmith** [comment]
Yes, but it doesn't work against a task being completed with from a ```tasks query on a different file

**Comment by ivan-lednev** [comment]
> Yes, but it doesn't work against a task being completed with from a ```tasks query on a different file

This would likely require a change in the obsidian-tasks plugin. It might take significant effort, so marking this as `maybe/some day`

**Comment by FearTheBadger** [comment]
I found an acceptable way around this.

There is a plugin based on chron call `obsidian-chron`. You can set it via chron timing to run a command. I have mine running every 15 minutes (`*/15 * * * *`) running the `Archive tasks in this file` command.

https://github.com/cdloh/obsidian-cron

**Comment by Raijin56** [comment]
> I found an acceptable way around this.
> 
> There is a plugin based on chron call `obsidian-chron`. You can set it via chron timing to run a command. I have mine running every 15 minutes (`*/15 * * * *`) running the `Archive tasks in this file` command.
> 
> https://github.com/cdloh/obsidian-cron

Hi FearTheBadger,

Do you or someone knows how to make obsidian-cron to run the command only on a specific file? I'd like to run it in my Tasks file but not in all the files that contain tasks. I'm afraid the lack of a field in obsidian-cron to fill with the file in which to run the command would do that...

**Comment by FearTheBadger** [comment]
There is supposedly an API endpoint on obsidian-chron. But I can't get it to register anything useful, that would solve your problem.

**Comment by medo0o0o0** [comment]
Yes please +1
